src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -2304,4 +2428,67 @@ public UIDispatcherEventArgs(Action a)
         /// </summary>
         public Action ActionToDispatch { get; set; }
     }
+
+    /// <summary>
+    /// The LacingStrategyConverter is used to serialize and deserialize LacingStrategy enum values.
+    /// The mapping to string like 'applyDisabled' is to support the historical representation
+    /// of 'lacing' on Flood.
+    /// </summary>
+    public class LacingStrategyConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(LacingStrategy);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var s = reader.Value.ToString();
+            switch (s)
+            {
+                case "applyAuto":
+                    return LacingStrategy.Auto;
+                case "applyCartesianProduct":
+                    return LacingStrategy.CrossProduct;
+                case "applyDisabled":
+                    return LacingStrategy.Disabled;
+                case "applyFirst":
+                    return LacingStrategy.First;
+                case "applyLongest":
+                    return LacingStrategy.Longest;
+                case "applyShortest":
+                    return LacingStrategy.Shortest;
+                default:
+                    return LacingStrategy.Disabled;
+            }
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
```

Comment:
This class inherits from `JsonConverter` and is used during automatic-deserialization. These methods need to be public because they override the methods on the base class. But, they are never called by the user. They are only called by the Json deserializer. Their functionality is documented in Json.net's documentation. Can I let these slide?